### PR TITLE
Implement RS0037: Public API files must exist

### PR DESCRIPTION
**Pull request by Digmaster**
Implement analyzer for https://github.com/dotnet/roslyn-analyzers/issues/2615

Uses ID RS0037, which is not currently in use (Another PR uses RS0036).

This does not include a codefix.

### CODE REVIEW

File: src/PublicApiAnalyzers/UnitTests/DeclarePublicApiAnalyzerTests.cs [hunk 1]
```diff
@@ -118,6 +136,54 @@ private async Task VerifyAdditionalFileFixAsync(string language, string source,
...
+        {
+            var source = @"
+public class C
+{
+    private C() { }
+}
+";
+
+            string shippedText = null;
+            string unshippedText = @"";
```
Comment (mavasani) [inline src/PublicApiAnalyzers/UnitTests/DeclarePublicApiAnalyzerTests.cs]: Can we add a test where both `shippedText` and `unshippedText` are empty files, and verify there is no diagnostic?

**Comment by mavasani** [comment]
@Digmaster Thanks for the contribution. Can you please file an issue to track the code fix?

File: src/Roslyn.Diagnostics.Analyzers/Core/RoslynDiagnosticIds.cs [hunk 2]
```diff
@@ -39,5 +39,6 @@ internal static class RoslynDiagnosticIds
         public const string ImportingConstructorShouldBeObsoleteRuleId = "RS0033";
         public const string ExportedPartsShouldHaveImportingConstructorRuleId = "RS0034";
         public const string RestrictedInternalsVisibleToRuleId = "RS0035";
+        public const string PublicApiFilesMissing = "RS0037";
```
Comment (mavasani) [inline src/Roslyn.Diagnostics.Analyzers/Core/RoslynDiagnosticIds.cs]: I would recommend you to use `RS0036` as yours is a more non-controversial PR which should go in sooner. The other PR should get a merge conflict and move to `RS0037`.

File: src/PublicApiAnalyzers/Core/Analyzers/DeclarePublicApiAnalyzer.cs [hunk 3]
```diff
@@ -97,6 +97,16 @@ public sealed partial class DeclarePublicApiAnalyzer : DiagnosticAnalyzer
...
             customTags: WellKnownDiagnosticTags.Telemetry);
 
+        internal static readonly DiagnosticDescriptor PublicApiFilesMissing = new DiagnosticDescriptor(
+            id: DiagnosticIds.PublicApiFilesMissing,
+            title: PublicApiAnalyzerResources.PublicApiFilesMissingTitle,
+            messageFormat: PublicApiAnalyzerResources.PublicApiFilesMissingMessage,
+            category: "ApiDesign",
+            defaultSeverity: DiagnosticHelpers.DefaultDiagnosticSeverity,
+            isEnabledByDefault: DiagnosticHelpers.EnabledByDefaultIfNotBuildingVSIX,
+            helpLinkUri: @"https://github.com/dotnet/roslyn-analyzers/blob/master/src/PublicApiAnalyzers/PublicApiAnalyzers.Help.md",
```
Comment (mavasani) [inline src/PublicApiAnalyzers/Core/Analyzers/DeclarePublicApiAnalyzer.cs]: Thanks for the helpLink!

**Comment by Digmaster** [comment]
> 
> 
> @Digmaster Thanks for the contribution. Can you please file an issue to track the code fix?

Done!
https://github.com/dotnet/roslyn-analyzers/issues/2628
